File: 1358/solution.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        di = {'a':0, 'b':0, 'c':0}
        res = 0
        l = 0 
        for r in range(len(s)):
            di[s[r]]+=1
            while di['a']>=1 and di['b']>=1 and di['c']>=1:
                res+=(len(s)-r)
                di[s[l]]-=1
                l+=1
        return res

File: 1358/solution.java
class Solution {
    public int numberOfSubstrings(String s) {
        int[] arr = {0,0,0};
        int i, l=0, n=s.length(), res=0;
        for(int r=0;r<n;r++){
            i = (int) s.charAt(r)-97; 
            arr[i]++;
            while(arr[0]>=1 && arr[1]>=1 && arr[2]>=1){
                res+=(n-r);
                i = (int) s.charAt(l)-97;
                arr[i]--;
                l++; 
            }
        }
        return res;
    }
}
